### PR TITLE
Apm 1750 - refactor existing tests to use on demand test apps

### PR DESCRIPTION
**Pull request by luckmanali**
## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: api_tests/scripts/config.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from os import environ
+
+
+def _get_service_name():
```
Comment (Sourceless) [inline api_tests/scripts/config.py:4]: This looks like it isn't being used

File: api_tests/scripts/generic_request.py [hunk 2]
```diff
@@ -14,17 +13,11 @@ class GenericRequest:
...
         """Verify the arguments and then send a request and return the response"""
-        try:
-            url = self.endpoints[endpoint]
-        except KeyError:
-            if self.is_url(endpoint):
-                url = endpoint
-            else:
-                raise Exception("Endpoint not found")
+        if not self.is_url(url):
+            raise Exception("Endpoint not found")
```
Comment (Sourceless) [inline api_tests/scripts/generic_request.py:20]: Would suggest `RuntimeError`, using broad `Exception` can sometimes complicate catching errors nicely when a few layers are involved.

File: api_tests/tests/conftest.py [hunk 3]
```diff
@@ -64,17 +67,122 @@ def helper():
...
+    await app.destroy_app()
 
-    oauth = OauthHelper(config.CLIENT_ID, config.CLIENT_SECRET, config.REDIRECT_URI)
-    setattr(request.cls, "oauth", oauth)
 
-    yield  # Handover to test
+@pytest.yield_fixture(scope="class")
+def setup_user_restricted_app(request, test_app, test_product):
+    print("\nCreating User Restricted App..")
+    asyncio.run(test_app.create_new_app())
```
Comment (Sourceless) [inline api_tests/tests/conftest.py:104]: Could this setup be simplified a bit by making it async instead of sync?

**Review by Sourceless — CHANGES_REQUESTED** [review]
LGTM for the most part, just a few small bits

File: api_tests/tests/conftest.py (hunk 3 above)
Comment (luckmanali) [reply api_tests/tests/conftest.py:104]: i don't think you can make a class fixture async.

Comment (luckmanali) [reply api_tests/tests/conftest.py]: I have removed this bit of code as it not currently needed/used

**Review by Sourceless — APPROVED** [review]

**Review by Sourceless — APPROVED** [review]

**Pull request merged**